### PR TITLE
refactor: DPTP-1593 opensearch provider

### PR DESCRIPTION
**Pull request by DeveloperConnor**
Addresses [DBTP-1593](https://uktrade.atlassian.net/browse/DBTP-1593)

---
### Title:
- [ ] Scope included as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Ticket reference included (unless it's a quick out of ticket thing)
### Description:
- [ ] Link to ticket included (unless it's a quick out of ticket thing)
- [ ] Includes tests (or an explanation for why it doesn't)
- [ ] If the work includes user interface changes, before and after screenshots included in description
- [ ] Includes any applicable changes to the documentation in this code base
- [ ] Includes link(s) to any applicable changes to the documentation in the [DBT Platform Documentation](https://platform.readme.trade.gov.uk/) (can be to a pull request)
### Tasks:
- [ ] [Run the end to end tests for this branch]([https://github.com/uktrade/platform-tools?tab=readme-ov-file#regression-tests](https://github.com/uktrade/platform-end-to-end-tests?tab=readme-ov-file#running-the-tests)) and confirm that they are passing

https://eu-west-2.console.aws.amazon.com/codesuite/codepipeline/pipelines/willg-end-to-end-tests/executions/db057fb5-8bfd-4cda-8758-aa8e8bc46983/visualization?region=eu-west-2

[DBTP-1593]: https://uktrade.atlassian.net/browse/DBTP-1593?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dbt_platform_helper/providers/opensearch.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            ]
+
+            cache_provider.update_cache("opensearch", supported_versions)
+
+            return supported_versions
+
+        else:
+            return cache_provider.read_supported_versions_from_cache("opensearch")
+
+    # TODO - cache provider instantiated here rather than via dependancy injection since it will likely only be used in the get_supported_redis_versions method.
```
Comment (ksugden) [inline dbt_platform_helper/providers/opensearch.py:32]: I guess this comment needs to be updated to `get_supported_opensearch_versions` function

File: dbt_platform_helper/providers/opensearch.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+from dbt_platform_helper.providers.cache import CacheProvider
+
+
+class OpensearchProvider:
+
+    def __init__(self, opensearch_client):
+        self.openserarch_client = opensearch_client
```
Comment (a-gleeson) [inline dbt_platform_helper/providers/opensearch.py:7]: typo

File: dbt_platform_helper/providers/opensearch.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    def get_supported_opensearch_versions(self):
```
Comment (a-gleeson) [inline dbt_platform_helper/providers/opensearch.py:9]: should we include the return type in method signature?

**Review by a-gleeson — APPROVED** [review]
LGTM, just a couple small comments

**Pull request merged**